src/core/src/pass/constant_folding.cpp

```diff
@@ -105,6 +105,17 @@ bool ov::pass::ConstantFolding::run_on_model(const std::shared_ptr<ov::Model>& m
 
     for (const auto& original_node : model->get_ordered_ops()) {
         auto node = original_node;
+        if (!original_node->can_constant_fold(original_node->input_values())) {
+            if (auto sub_graph_node = std::dynamic_pointer_cast<ov::op::util::MultiSubGraphOp>(node)) {
```

Comment:
Maybe just for all sub-graph run constant folding. Then if all folded I assume the sub-graph node will also fold when calling `node->constant_fold` .